src/config.rs

```diff
@@ -166,6 +167,26 @@ impl IndexMut<usize> for ColorList {
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct Shell<'a> {
+    program: Cow<'a, str>,
+    args: Vec<String>,
+}
+
+impl<'a> Shell<'a> {
+    pub fn new(program: &'a str) -> Shell<'a> {
```

Comment:
By the way, if you end up working on the CLI parsing part of this, I've asked about how to make it work with `clap` since I had no idea:

```
17:46 <kbknapp> You can use `Arg::allow_hyphen_values(true)` and `Arg::multiple(true)` on the `-e` arg which should accomplish what you want. Note that this won't force -e to be the last option, but it will treat -e as the last option, as everything after `-e` will be treated as a value for `-e`
17:47 <kbknapp> Here's the docs page for `allow_hyphen_values` https://docs.rs/clap/2.20.0/clap/struct.Arg.html#method.allow_hyphen_values
17:58 <kbknapp> Here's a playground example: http://play.integer32.com/?gist=aa29abe77a13a3f586935336fe1613ee&version=stable
```